### DISCUSSION TITLE
Unable to publish content

### DISCUSSION THREAD

**Discussion by paritshivani** [Sunbird-Knowlg/Community#57] [General Discussions]
Hi @keshavprasadms, @vinukumar-vs @AmiableAnil

Hope you are doing well!..

Site Environments Details - SunbirdEd portal of version 4.4.1

Issue - We are unable to Publish few content getting following error.

API Details -

Request URL: host//content/content/v1/publish/do_id
Status Code: 400 Bad Request

Payload -
{
    "request": {
        "content": {
            "publishChecklist": [
                "No Hate speech, Abuse, Violence, Profanity",
                "No Sexual content, Nudity or Vulgarity",
                "Appropriate Title, Description",
                "Correct Category, Sub-Category, Topic, Language",
                "Content plays correctly",
                "Can see the content clearly on Desktop and App",
                "No Discrimination or Defamation",
                "Is suitable for all stakeholders/ scholars",
                "Appropriate tags such as Resource Type, Concepts",
                "Relevant Keywords",
                "Audio (if any) is clear and easy to understand",
                "No Spelling mistakes in the text",
                "Language is simple to understand"
            ],
            "lastPublishedBy": "6939db9e-94ac-4748-b53f-7b2fc59b30b1"
        }
    }
}
Response -
{
    "id": "api.content.publish",
    "ver": "1.0",
    "ts": "2022-09-02T14:16:37.075Z",
    "params": {
        "resmsgid": "db154230-2ac9-11ed-a831-93acf1dda9a5",
        "msgid": null,
        "status": "failed",
        "err": "MISSING_ASSETS",
        "errmsg": "Error! Missing Asset. | [Asset Id 'b7ab156d-6cc0-47b9-b3c8-0721a05913ba' is missing.]"
    },
    "responseCode": "CLIENT_ERROR",
    "result": {}
}
Can you please look into it and let us know the steps to resolve the same.

cc @coolbung, @Renuka-S, @vinod303

**Comment by krgauraw** [comment]
@paritshivani Publish api try to download all asset including plugins. So please check if below configuration in learning-service (vm based application) is having correct value or not. 

content.media.base.url="{{content_media_base_url}}"
plugin.media.base.url="{{plugin_media_base_url}}"

variable values should be as below:
plugin_media_base_url: "{{proto}}://{{domain_name}}"
content_media_base_url: "{{proto}}://{{domain_name}}"

cc: @vinukumar-vs @AmiableAnil

**Reply by Renuka-S** [reply]
@saiakhil46 Please check. cc: @paritshivani @coolbung

**Reply by paritshivani** [reply]
@krgauraw , Above configuratons are correct but still facing the same issue

`{"id":"api.content.publish","ver":"1.0","ts":"2022-09-06T10:18:24.003Z","params":{"resmsgid":"3d668d30-2dcd-11ed-a831-93acf1dda9a5","msgid":null,"status":"failed","err":"MISSING_ASSETS","errmsg":"Error! Missing Asset. | [Asset Id '5cbb5d15-10e5-4a15-ac78-6e91995962b1' is missing.]"},"responseCode":"CLIENT_ERROR","result":{}}`

Any other thing need to checked?

@Renuka-S , @vinod303, @saiakhil46, @coolbung

**Reply by AmiableAnil** [reply]
@paritshivani Please share the learning service log.

**Reply by saiakhil46** [reply]
@AmiableAnil Initially we see below unknownhost error
`{"eid":"ERROR","ets":1662641738569,"ver":"3.0","mid":"LP.1662641738569.879632db-ea2e-453b-abd7-a7c358e743af","actor":{"id":"org.sunbird.learning.platform","type":"User"},"context":{"channel":"0130171255884513283","pdata":{"id":"dev.dev.learning.platform","pid":"learning-service","ver":"1.0"},"env":"content"},"edata":{"err":"SERVER_ERROR","stacktrace":"java.net.UnknownHostException: nuisstaging.blob.core.windows.net\n\tat `

I observed that no host is accessble from learning-service vm, so i restarted the vm and unknownhost issue got resolved.

After that there are some SSLhandshake error as below.

`{"eid":"ERROR","ets":1662645071011,"ver":"3.0","mid":"LP.1662645071011.a4d88b19-c968-44cc-8010-1871ec96e782","actor":{"id":"org.sunbird.learning.platform","type":"User"},"context":{"channel":"in.ekstep","pdata":{"id":"dev.dev.learning.platform","pid":"learning-service","ver":"1.0"},"env":"system"},"edata":{"err":"SERVER_ERROR","stacktrace":"javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target\n\tat sun.security.ssl.Alerts.getSSLException(Alerts.java:192)\n\tat sun.security.ssl.SSLSocketImpl.fatal(SSLSocketImpl.java:1949)\n\tat sun.security.ssl.Handshaker.fatalSE(Handshaker.java:302)\n\tat sun.security.ssl.Handshaker.fatalSE(Handshaker.java:296)\n\tat sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1514)\n\tat sun.security.ssl.ClientHandshaker.processMessage(ClientHandshaker.java:216)\n\tat sun.security.ssl.Handshaker.processLoop(Handshaker.java:1026)\n\tat sun.security.ssl.Handshaker.process_record(Handshaker.java:961)\n\tat sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:1062)\n\tat sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1375)\n\tat sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1403)\n\tat sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1387)\n\tat sun.net.www.protocol.https.HttpsClient.afterConnect(HttpsClient.java:559)\n\tat sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:185)\n\tat sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1546)\n\tat sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1474)\n\tat java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:480)\n\tat sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:338)\n\tat org.sunbird.common.util.HttpDownloadUtility.downloadFile(HttpDownloadUtility.java:43)\n\tat org.sunbird.content.concrete.processor.LocalizeAssetProcessor$1.call(LocalizeAssetProcessor.java:211)\n\tat org.sunbird.content.concrete.processor.LocalizeAssetProcessor$1.call(LocalizeAssetProcessor.java:189)\n\tat java.util.concurrent.FutureTask.run(FutureTask.java:266)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)\n\tat java.lang.Thread.run(Thread.java:745)\nCaused by: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target\n\tat sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:387)\n\tat sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:292)\n\tat sun.security.validator.Validator.validate(Validator.java:260)\n\tat sun.security.ssl.X509TrustManagerImpl.validate(X509TrustManagerImpl.java:324)\n\tat sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:229)\n\tat sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:124)\n\tat sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1496)\n\t... 20 more\nCaused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target\n\tat sun.security.provider.certpath.SunCertPathBuilder.build(SunCertPathBuilder.java:141)\n\tat sun.security.provider.certpath.SunCertPathBuilder.engineBuild(SunCertPathBuilder.java:126)\n\tat java.security.cert.CertPathBuilder.build(CertPathBuilder.java:280)\n\tat sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:382)\n\t... 26 more\n","errtype":"system"},"syncts":1662645071011}`

For this i have updated java version from **1.8.0_121** to **1.8.0_342** after the SSL issue got fixed.

**Reply by saiakhil46** [reply]
@AmiableAnil Now the publishing issue got resolved. Thanks for the support
